addons/mail/models/mail_activity.py

```diff
@@ -795,3 +796,130 @@ def activity_unlink(self, act_type_xmlids, user_id=None):
             domain = ['&'] + domain + [('user_id', '=', user_id)]
         self.env['mail.activity'].search(domain).unlink()
         return True
+
+    def _read_progress_bar(self, domain, group_by, progress_bar):
+        if not (progress_bar['field'] == 'activity_state' and self._fields[group_by].store):
+            return super()._read_progress_bar(domain, group_by, progress_bar)
+
+        # use datetime.utcnow() instead of date.today(), because
+        # it's used in _compute_state_from_date and on mocking in tests
+        today_utc = datetime.utcnow()
+        today = date(year=today_utc.year, month=today_utc.month, day=today_utc.day)
```

Comment:
Pure technical suggestion, the `datetime` module is hard to mock in the unittests and we have an internal function that does exactly what you want :)
```suggestion
        today = fields.Date.context_today(self.with_context(tz='UTC'))
```

Reply:
Yes, but the tests  already mock datetime and using ``context_today`` may break it, no?

https://github.com/odoo/odoo/blob/62c849521ef06aeff212cbdd7179a49155276233/addons/test_mail/tests/test_mail_activity.py#L375-L378

https://github.com/odoo/odoo/blob/6c4623a46cbe35c18017c59335b3a3c563ae5fbf/odoo/fields.py#L1650-L1672

Reply:
`today_utc = datetime.utcnow().date()` is much simpler, IMO

Reply:
@Elkasitu yes, but I'm doing a bugfix, so I copy-pasted this code without changes

https://github.com/odoo/odoo/blob/6c4623a46cbe35c18017c59335b3a3c563ae5fbf/addons/mail/models/mail_activity.py#L189-L191